charts/redpanda/values.yaml

```diff
@@ -459,12 +459,25 @@ storage:
       # -- Additional annotations to apply to the created PersistentVolumeClaims.
       annotations: {}
 
-    # credentialsSecretRef can be used to set cloud_storage_secret_key from
+    # credentialsSecretRef can be used to set `cloud_storage_secret_key` and/or `cloud_storage_access_key` from
```

Comment:
This way of setting values is quite confusing to me. Wouldn't it be more ergonomic, both for code and documentation, to allow the tiered_storage settings to be either a static value or a secret ref object?

Better yet, could we just expose the envvar configuration and let users control `valueFrom`?

Reply:
Redpanda has 3 different type of configuration.
1. [Cluster](https://docs.redpanda.com/current/reference/cluster-properties/)
2. [Node](https://docs.redpanda.com/current/reference/node-properties/)
3. [Tunable](https://docs.redpanda.com/current/reference/tunable-properties/)

## Restart required

If user would change node configuration that would require full rollout for the StatefulSet controller. The checksum would need to change every time the referenced secret changes 👇  https://github.com/redpanda-data/helm-charts/blob/a4d526f692c11311270582d75c23cb222513484c/charts/redpanda/templates/statefulset.yaml#L57

## Admin API config changes

The tunable and cluster configuration does not need to have full rollout, but post-update job needs to be re-triggered 👇 
https://github.com/redpanda-data/helm-charts/blob/a4d526f692c11311270582d75c23cb222513484c/charts/redpanda/templates/post-upgrade.yaml#L73

User would need to run `rpk cluster import` every time the secret content changes or re-run `post-upgrade` job.

## Dynamic configuration

Few configuration properties that relay on the Pod being scheduled (e.g. Pod ordinal) will be adjusted in initContainer 👇 
https://github.com/redpanda-data/helm-charts/blob/a4d526f692c11311270582d75c23cb222513484c/charts/redpanda/templates/statefulset.yaml#L144-L172

The script is defined 👇 
https://github.com/redpanda-data/helm-charts/blob/a4d526f692c11311270582d75c23cb222513484c/charts/redpanda/templates/secrets.yaml#L288-L359 

## Size of the Volume

The `storage_min_free_bytes` depends on `storage.persistentVolume.size` which is not part of the configuration, but dictates that buffer for Redpanda configuration, so that in the end we don't see alloc failures.  

## Changes to K8S resources

We can support such configuration, but some places that uses configuration properties changes other K8S resource like listed below 👇 :

- https://github.com/redpanda-data/helm-charts/blob/a4d526f692c11311270582d75c23cb222513484c/charts/redpanda/templates/statefulset.yaml#L486
- https://github.com/redpanda-data/helm-charts/blob/a4d526f692c11311270582d75c23cb222513484c/charts/redpanda/templates/statefulset.yaml#L459 the `(include "storage-tiered-config" .|fromJson).cloud_storage_enabled` part
- https://github.com/redpanda-data/helm-charts/blob/a4d526f692c11311270582d75c23cb222513484c/charts/redpanda/templates/statefulset.yaml#L385
- https://github.com/redpanda-data/helm-charts/blob/a4d526f692c11311270582d75c23cb222513484c/charts/redpanda/templates/statefulset.yaml#L373
- https://github.com/redpanda-data/helm-charts/blob/a4d526f692c11311270582d75c23cb222513484c/charts/redpanda/templates/statefulset.yaml#L298
- https://github.com/redpanda-data/helm-charts/blob/a4d526f692c11311270582d75c23cb222513484c/charts/redpanda/templates/statefulset.yaml#L254
- https://github.com/redpanda-data/helm-charts/blob/a4d526f692c11311270582d75c23cb222513484c/charts/redpanda/templates/statefulset.yaml#L166
- https://github.com/redpanda-data/helm-charts/blob/a4d526f692c11311270582d75c23cb222513484c/charts/redpanda/templates/statefulset.yaml#L126
- https://github.com/redpanda-data/helm-charts/blob/a4d526f692c11311270582d75c23cb222513484c/charts/redpanda/templates/_configmap.tpl#L616
- https://github.com/redpanda-data/helm-charts/blob/a4d526f692c11311270582d75c23cb222513484c/charts/redpanda/templates/_helpers.tpl#L334
- https://github.com/redpanda-data/helm-charts/blob/a4d526f692c11311270582d75c23cb222513484c/charts/redpanda/templates/post-upgrade.yaml#L74-L93
- https://github.com/redpanda-data/helm-charts/blob/a4d526f692c11311270582d75c23cb222513484c/charts/redpanda/templates/_configmap.tpl#L190-L192
- https://github.com/redpanda-data/helm-charts/blob/a4d526f692c11311270582d75c23cb222513484c/charts/redpanda/templates/_configmap.tpl#L90-L92